acs-integration-tests/src/test/java/com/ge/predix/cloud/foundry/application/ApplicationHelper.java

```diff
@@ -0,0 +1,353 @@
+            throws InterruptedException {
+
+        pushApplication(this.cloudFoundryOperations, applicationName, application)
+                .thenMany(setEnvironmentVariables(this.cloudFoundryOperations, applicationName, environmentVariables))
+                .thenMany(createServiceInstances(this.cloudFoundryOperations, services))
+                .thenMany(bindServiceInstances(this.cloudFoundryOperations, applicationName,
+                        services.stream().map(CloudFoundryService::getServiceInstanceName)
+                                .collect(Collectors.toList())))
+                .blockLast();
+    }
+
+    private static Mono<Void> pushApplication(final CloudFoundryOperations cloudFoundryOperations,
+            final String applicationName, final Path application) {
+
+        return cloudFoundryOperations.applications()
+                .push(PushApplicationRequest.builder()
+                        .application(application)
+                        .host(applicationName)
+                        .domain(System.getProperty("CF_BASE_DOMAIN"))
+                        .buildpack("java-buildpack")
+                        .diskQuota(2048)
+                        .healthCheckType(ApplicationHealthCheck.PORT)
+                        .memory(2048)
+                        .name(applicationName)
+                        .noStart(true)
+                        .build())
+                .doOnError(onError -> LOGGER.error("Unable to push application '{}'", applicationName));
+    }
+
+    void setEnvironmentVariables(final String applicationName, final Map<String, String> environmentVariables)
+            throws InterruptedException {
+
+        setEnvironmentVariables(this.cloudFoundryOperations, applicationName, environmentVariables)
+                .blockLast();
+    }
+
+    private static Flux<Void> setEnvironmentVariables(final CloudFoundryOperations cloudFoundryOperations,
+            final String applicationName, final Map<String, String> environmentVariables) {
+
+        return Flux.fromIterable(environmentVariables.entrySet())
+                .flatMap(environmentVariable -> setEnvironmentVariable(cloudFoundryOperations, applicationName,
+                        environmentVariable.getKey(), environmentVariable.getValue()), 1);
+    }
+
+    private static Mono<Void> setEnvironmentVariable(final CloudFoundryOperations cloudFoundryOperations,
+            final String applicationName, final String variableName, final String variableValue) {
+
+        return cloudFoundryOperations.applications()
+                .setEnvironmentVariable(SetEnvironmentVariableApplicationRequest.builder()
+                        .name(applicationName)
+                        .variableName(variableName)
+                        .variableValue(variableValue)
+                        .build())
+                .doOnError(onError -> LOGGER.error("Unable to set environment variable '{}': '{}' on '{}'",
```

Comment:
In theory this should fail the suite but maybe the `Throwable` is getting lost in translation. Will have to write a meta-test to actually force an in-flux exception and verify.